longtailstorelib/gcsstore.go

```diff
@@ -120,16 +120,17 @@ func (blobObject *gcsBlobObject) Read() ([]byte, error) {
 }
 
 func (blobObject *gcsBlobObject) LockWriteVersion() (bool, error) {
-	blobObject.writeCondition = &storage.Conditions{DoesNotExist: true}
```

Comment:
This actually set the writeCondition if the file did not exists, but it did not clear the write condition if there was an error.
Restructured it a bit to make it clearer.